osutil/io.go

```diff
@@ -91,13 +92,29 @@ func AtomicWriteFileChown(filename string, data []byte, perm os.FileMode, flags
 		return errors.New("internal error: AtomicWriteFileChown needs none or both of uid and gid set")
 	}
 
-	if err := fd.Sync(); err != nil {
-		return err
+	if shouldReallySync {
+		if err := fd.Sync(); err != nil {
+			return err
+		}
 	}
 
 	if err := os.Rename(tmp, filename); err != nil {
 		return err
 	}
 
-	return dir.Sync()
+	if shouldReallySync {
+		return dir.Sync()
+	}
+	return nil
+}
+
+// Allow disabling sync for testing. This brings massive improvements on
+// certain filesystems (like btrfs) and very much noticeable improvements in
+// all unit tests in genreal.
+var shouldReallySync bool = true
```

Comment:
Let's please rename this to snapdUnsafeIO so the relationship to the variable is clear (we need to invert the tests too). We can also drop the = true/false part as it defaults to false.

In fact, we can probably inline:
```
var snapdUsafeIO = len(os.Args) > 0 && strings.HasSuffix(os.Args[0], ".test") && GetenvBool("SNAPD_UNSAFE_IO")